test/cross_serialization/FastCDRSerialization.cpp

```diff
@@ -14,9 +14,142 @@
 
 #include "FastCDRSerialization.hpp"
 
-TEST_F(FastCDRSerialization, double_sequence_with_aligment_serialization)
-{
-    double_sequence_with_aligment_serialization();
+using MyTypes = ::testing::Types<char, uint8_t, uint16_t, uint32_t, uint64_t, int8_t, int16_t, int32_t, int64_t, float,
+                double>;
+TYPED_TEST_SUITE(FastCDRSerialization, MyTypes);
+
+TYPED_TEST(FastCDRSerialization, micro_to_fast) {
+
+    for (size_t i = 1; i < this->byte_buffer.size(); i++)
+    {
+        typename TestFixture::partial_buffer split_buffers = this->split_buffer({i, this->byte_buffer.size() - i});
+        this->prepare_micro_cdr(split_buffers, [&](ucdrBuffer& cdr)
+                {
+                    TypeParam fill = 0;
+                    for (size_t k = 0; k < ELEMENT_NUMBER; k++)
+                    {
+                        EXPECT_TRUE(this->ucdr_serialize(cdr, fill++));
+                    }
+                });
+
+        this->prepare_fast_cdr([](Cdr& cdr)
+                {
+                    TypeParam out;
+                    TypeParam compare = 0;
+                    for (size_t j = 0; j < ELEMENT_NUMBER; j++)
+                    {
+                        cdr.deserialize(out);
+                        ASSERT_EQ(compare++, out);
+                    }
+                });
+    }
 }
 
+// TYPED_TEST(FastCDRSerialization, micro_to_fast_multiple_slots) {
```

Comment:
There should be other 4 tests where the buffer is split in N blocks instead of in 2 blocks of (N, N-L) elements